File: tutorials/elixir/06-hello-structured-logging/lib/hello_logging.ex
defmodule HelloLogging do
  require Logger

  @doc """
  Demonstrates structured logging at different levels with metadata.
  """
  def demo do
    Logger.info("Processing order", order_id: 123, user_id: "abc")
    Logger.warning("Slow query", duration_ms: 1500)
    Logger.error("Connection failed", service: "database")
  end
end

File: tutorials/erlang/06-hello-structured-logging/src/hello_logging.erl
-module(hello_logging).
-export([demo/0]).

demo() ->
    logger:info("Processing order", #{order_id => 123, user_id => <<"abc">>}),
    logger:warning("Slow query", #{duration_ms => 1500}),
    logger:error("Connection failed", #{service => <<"database">>}).
